tester_app/client/app/lib/middlewares/loggerMiddleware.js

```diff
@@ -0,0 +1,25 @@
+/* eslint no-console: 0 */
+
+export default function logger({ getState }) {
+  return next => action => {
+    console.log('will dispatch', action);
+
+    // Call the next dispatch method in the middleware chain.
+    const result = next(action);
+
+    // We can't _read_ immutable objects in console out-of-the-box.
+    const immutableState = getState();
+    const readableState = {};
+    for (const storeItem in immutableState) {
```

Comment:
@alexfedoseev I think this could be much more readable using lodash.reduce

Like https://lodash.com/docs#reduce
```
_.reduce({ 'a': 1, 'b': 2 }, function(result, n, key) {
  result[key] = n * 3;
  return result;
}, {});
```
How's this (note, I didn't try this)

```javascript
import _ from 'lodash';



const readableState = _.reduce(immutableState, (result, immutable, key) => {
  result[key] = immutable.toJS();
}, {});
```